### PR TITLE
Feature/use yaml config#372

### LINKED ISSUE #372: Replace JSON Config with YAML
**Description**:
A personal bugbear of mine has been that the JSON config is a bit awkward to use during development. As you use different ICAT URLs to contact different ICAT stacks, you have to manually cut and paste new URLs. This is because in JSON you can't comment out lines of the file.

YAML files, in my opinion, serve as better config files. We use them in OperationsGateway API (see the [example config file](https://github.com/ral-facilities/operationsgateway-api/blob/main/operationsgateway_api/config.yml.example)) and you can have multiple sections (similar to nested JSON) as well as the ability to comment out specific lines.

**Acceptance criteria**:
- [ ] Check whether JSON config must remain for production (I can check with someone if that's easier)
- [x] Add YAML config functionality so `config.yaml` will be read and understood by the API
- [x] Provide an example YAML config file with the same settings as the existing one
- [x] If JSON config can be removed then remove that functionality
- [ ] If JSON config needs to stay then make sure both formats work. If both file formats are present then we should take one over the other
- [x] If JSON config gets removed, replace the steps in the GitHub Actions build that edit the config file to make it work with the new file format
- [x] Prepare YAML config files for DataGateway production and dev machines (for both dev and prod). We'll need to get hold of the existing config files (I can do that) so the DataGateway team have to do as little as possible

### PR DESCRIPTION
**Pull request by Reillyhewitson**
This PR will close #372 

## Description
Enter a description of the changes here

## Testing Instructions
Add a set up instructions describing how the reviewer should test the code

- [ ] Review code
- [ ] Check GitHub Actions build
- [ ] If `icatdb Generator Script Consistency Test` CI job fails, is this because of a deliberate change made to the script to change generated data (which isn't actually a problem) or is here an underlying issue with the changes made?
- [ ] Review changes to test coverage
- [ ] Does this change mean a new patch, minor or major version should be made? If so, does one of the commit messages feature `fix:`, `feat:` or `BREAKING CHANGE:` so a release is automatically made via GitHub Actions upon merge?
- [ ] {more steps here}

## Agile Board Tracking
Connect to #372

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -436,10 +436,12 @@ tests (containing tests for backend specific components, including tests for the
 different types of endpoints) and Database Backend tests (like the ICAT backend tests,
 but covering only the most used aspects of the API).
 
-The configuration file (`config.json`) contains two options that will be used during the
+The configuration file (`config.yaml`) contains two options that will be used during the
 testing of the API. Set `test_user_credentials` and `test_mechanism` appropriately for
-your test environment, using `config.json.example` as a reference. The tests require a
-connection to an instance of ICAT 5, so set the rest of the config as needed. These tests have been written to assume ICAT 5 has been installed so please update to that or test with an older version of the API.
+
```
Comment (MRichards99) [inline README.md:441]: This blank line looks to be causing some formatting issues (https://github.com/ral-facilities/datagateway-api/blob/1e08ed90d1243bd335590a42e93ef99c34a57e05/README.md#running-tests). Could you resolve that please (by removing the blank line)?

**Review by MRichards99 — CHANGES_REQUESTED** [review]
Looks good, just a couple of picky things that I've picked up on (sorry!).

File: noxfile.py [hunk 2]
```diff
@@ -72,7 +72,8 @@ def safety(session):
             f"--output={requirements.name}",
             external=True,
         )
-        # Ignore 50916 as the latest version of pydantic does not support
+        # Ignore 50916 as the latest version of pydantic and
+        # Ignore 5146 as the latest version of pytest does not support
```
[the remark this answers is not in the document]
Comment (Reillyhewitson) [reply noxfile.py]: Fixed in latest commit

File: .github/workflows/ci-build.yml [hunk 3]
```diff
@@ -84,14 +84,17 @@ jobs:
...
+      - name: Configure datagateway extension
+        run: echo "`yq \
+          '.datagateway_api.extension="/datagateway_api"' datagateway_api/config.yaml.example`" > datagateway_api/config.yaml.example
+      - name: Create config.yaml
+        run: cp datagateway_api/config.yaml.example datagateway_api/config.yaml
       - name: Create search_api_mapping.json
         run: cp datagateway_api/search_api_mapping.json.example datagateway_api/search_api_mapping.json
 
-      # Install Nox, Poetry and API's dependencies
+        # Install Nox, Poetry and API's dependencies
```
[the remark this answers is not in the document]
Comment (Reillyhewitson) [reply .github/workflows/ci-build.yml]: Fixed in latest commit

File: README.md (hunk 1 above)
Comment (Reillyhewitson) [reply README.md]: Fixed in latest commit

**Review by MRichards99 — APPROVED** [review]

**Pull request merged**